lib/dkim/dkim_header.rb

```diff
@@ -33,5 +33,54 @@ def encoder_for key
         raise "unknown key: #{key}"
       end.new
     end
+
+    def wrap(raw, prepend=0) # :nodoc:
```

Comment:
It seems that `wrap` responsibility is different than the DkimHeader. Can it be abstracted away into its own class  (i.e DkimWrapper)?
I like the Sandi Metz principle that methods shouldn't be longer than 5 lines. I also like this quote from Robert Martin:
> The first rule of functions is that they should be small. The second rule of functions is that they should be smaller than that.

Maintaining this code can be hard with so much going on. Could this method be refactored into multiple small methods in its own class?

Reply:
> Maintaining this code can be hard with so much going on. Could this method be refactored into multiple small methods in its own class?

I did write a nicer looking version, with several smaller functions, and it's about 2.5x slower. So, in the end, I reused the fast code that was already written.

Reply:
2.5x slower! That's a big performance hit

Reply:
@restebanez,  I mean. You can probably write cleaner just-as-fast code ; but that day I did not.

Reply:
@oz I have to agree with @restebanez , while it is definitely an absolutely amazing piece of Ruby code, it appears to be very difficult to understand and manage.

Could we try the slow version?